vendor/k8s.io/kubernetes/test/e2e/storage/drivers/csi.go

```diff
@@ -220,6 +219,9 @@ func (h *hostpathCSIDriver) PrepareTest(f *framework.Framework) (*testsuites.Per
 
 		ginkgo.By(fmt.Sprintf("deleting the driver namespace: %s", ns2))
 		tryFunc(deleteNamespaceFunc(f.ClientSet, ns2, framework.DefaultNamespaceDeletionTimeout))
+		// we need to remove the cleanupHandle at last so as if AfterEach and AfterSuite races with
+		// each other and AfterEach(or defer) run gets killed then it still cleanups.
+		framework.RemoveCleanupAction(h.cleanupHandle)
```

Comment:
This change in not in the referenced UPSTREAM PR. Upstream master still has `RemoveCleanupAction` as the first call in `cleanupFunc`.

Reply:
Yes. I dropped this in upstream PR. I intend to bring it back in a new PR in upstream. This is still needed.

Reply:
The please create a separate `UPSTREAM: <carry>` commit, so it does not get lost on the next rebase. fixing upstream first is even better.

Reply:
I updated this PR to use right commit from upstream.